apps/namadillo/src/atoms/proposals/atoms.ts

```diff
@@ -107,25 +112,29 @@ export const votedProposalIdsAtom = atomWithQuery((get) => {
   };
 });
 
-export const canVoteAtom = atomWithQuery((get) => {
-  const account = get(defaultAccountAtom);
-  const api = get(indexerApiAtom);
+export const canVoteAtom = atomFamily((proposalStartEpoch: bigint) =>
+  atomWithQuery((get) => {
+    const account = get(defaultAccountAtom);
+    const api = get(indexerApiAtom);
 
-  return {
-    queryKey: ["can-vote"],
-    enabled: account.isSuccess,
-    queryFn: async () => {
-      const all_bonds = await api.apiV1PosBondAddressGet(account.data!.address);
+    return {
+      queryKey: ["can-vote"],
```

Comment:
Not introduced in this PR, but `queryKey` is missing the dependencies (definitely `account` and I think possibly `api` too), so this doesn't react when the default account is changed in the extension.

Reply:
Cool! I will fix that